AppController/lib/hermes_client.rb

```diff
@@ -0,0 +1,134 @@
+    rescue Errno::ECONNREFUSED
+      raise FailedNodeException.new("Failed to call Hermes: connection refused")
+    end
+  end
+
+  # Gets haproxy statistics from Hermes located on load balancer node.
+  #
+  # Args:
+  #   lb_ip: IP address of load balancer node.
+  #   secret: Deployment secret.
+  #   fetch_servers: Determins if backend servers list should be fetched
+  #
+  def self.get_proxies_stats(lb_ip, secret, fetch_servers=true)
+    data = {
+      'include_lists' => {
+        'proxy' => ['name', 'accurate_frontend_scur', 'backend',
+                    'frontend', 'servers'],
+        'proxy.frontend' => ['req_tot'],
+        'proxy.backend' => ['qcur']
+      },
+      'max_age' => 0
+    }
+    if fetch_servers
+      data['include_lists']['proxy.server'] = [
+        'private_ip', 'port', 'status'
+      ]
+    end
+    proxies_list = HermesClient.make_call(
+      lb_ip, secret, '/stats/local/proxies', data
+    )
+    return proxies_list['proxies_stats']
+  end
+
+  # Gets haproxy statistics for a specific proxy
+  # from Hermes located on load balancer node.
+  #
+  # Args:
+  #   lb_ip: IP address of load balancer node.
+  #   secret: Deployment secret.
+  #   proxy_name: Name of proxy to return.
+  #   fetch_servers: Determins if backend servers list should be fetched
+  #
+  def self.get_proxy_stats(lb_ip, secret, proxy_name, fetch_servers=true)
+    proxies = HermesClient.get_proxies_stats(lb_ip, secret, fetch_servers)
+    proxy = proxies.detect{|item| item['name'] == proxy_name}
+    return proxy unless proxy.nil?
+    raise AppScaleException.new("Proxy #{proxy_name} was not found at #{lb_ip}")
+  end
+
+  # Gets total_requests, total_req_in_queue and current_sessions 
+  # for a specific proxy from Hermes located on load balancer node.
+  #
+  # Args:
+  #   lb_ip: IP address of load balancer node.
+  #   secret: Deployment secret.
+  #   proxy_name: Name of proxy to return.
+  #
+  def self.get_proxy_load_stats(lb_ip, secret, proxy_name)
+    proxy = HermesClient.get_proxy_stats(lb_ip, secret, proxy_name, false)
+    total_requests_seen = proxy['frontend']['req_tot']
+    total_req_in_queue = proxy['backend']['qcur']
+    current_sessions = proxy['accurate_frontend_scur']
+    Djinn.log_debug("HAProxy load stats for #{proxy_name} at #{lb_ip}: " \
+      "req_tot=#{total_requests_seen}, qcur=#{total_req_in_queue}, " \
+      "scur=#{current_sessions}")
+    return total_requests_seen, total_req_in_queue, current_sessions
+  end
+
+  # Gets running and failed backend servers for a specific proxy.
+  #
+  # Args:
+  #   lb_ip: IP address of load balancer node.
+  #   secret: Deployment secret.
+  #   proxy_name: Name of proxy to return.
+  #
+  def self.get_backend_servers(lb_ip, secret, proxy_name)
+    proxy = HermesClient.get_proxy_stats(lb_ip, secret, proxy_name, true)
+
+    running = proxy['servers'] \
+      .select{|server| not server['status'].start_with?('DOWN')} \
```

Comment:
@obino , I decided to keep old logic here to no dig into reasons why it was done so. But it's good point, do you know why previously we were verifying "not down", instead of "is up"?